### PR TITLE
Added Auto Scroll

### LINKED ISSUE #3874: [GSoC Task] Add pagination in the leaderboard screen to load more users on scrolling down.
**Summary:** 
Add pagination in the leaderboard screen to load more users on scrolling down.

- [x] Implement Pagination

- [x] Add Button for AutoScroll to user’s rank

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by madhurgupta10**
**Description**
Fixes #3874 

Added Auto-Scroll to My Rank feature

**Tests performed**
Tested betaDebug on Pixel 2 with API level 29

**Screenshots**
Screenshot_1598611213

### CODE REVIEW

**Comment by nicolas-raoul** [comment]
Wonderful!
The code is clean, I am ready to merge, but just a small suggestion: how about showing rank+1, so that the "My rank" button does not overlap my username?

**Comment by madhurgupta10** [comment]
> Wonderful!
> The code is clean, I am ready to merge, but just a small suggestion: how about showing rank+1, so that the "My rank" button does not overlap my username?

Sure I will do that :)

**Comment by madhurgupta10** [comment]
@nicolas-raoul Done :)

File: app/src/main/java/fr/free/nrw/commons/profile/leaderboard/LeaderboardFragment.java [hunk 1]
```diff
@@ -172,14 +172,15 @@ private void refreshLeaderboard() {
 
     /**
      * Performs Auto Scroll to the User's Rank
+     * We use userRank+1 to load one extra user and prevent overlapping of my rank button
      */
     private void scrollToUserRank() {
-        if (Objects.requireNonNull(leaderboardListRecyclerView.getAdapter()).getItemCount() > userRank) {
-            leaderboardListRecyclerView.smoothScrollToPosition(userRank);
+        if (Objects.requireNonNull(leaderboardListRecyclerView.getAdapter()).getItemCount() > userRank+1) {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/profile/leaderboard/LeaderboardFragment.java:178]: sorry for the very minor detail, but would you mind making these ` + 1`?
I mean, spaces around the `+` sign, that's the convention in Java :-)

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/profile/leaderboard/LeaderboardFragment.java (hunk 1 above)
Comment (madhurgupta10) [reply app/src/main/java/fr/free/nrw/commons/profile/leaderboard/LeaderboardFragment.java:178]: Sure!

**Comment by nicolas-raoul** [comment]
It works well in all orientations, thanks!

**Pull request merged**